packages/contracts-bedrock/scripts/deploy/UpgradeOPChain.s.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Script } from "forge-std/Script.sol";
+import { OPContractsManager } from "src/L1/OPContractsManager.sol";
+import { BaseDeployIO } from "scripts/deploy/BaseDeployIO.sol";
+
+contract UpgradeOPChainInput is BaseDeployIO {
+    address internal _prank;
+    OPContractsManager internal _opcm;
+    bytes _opChainConfigs;
+
+    // Setter for OPContractsManager type
+    function set(bytes4 _sel, address _value) public {
+        require(address(_value) != address(0), "UpgradeOPCMInput: cannot set zero address");
+
+        if (_sel == this.prank.selector) _prank = _value;
+        else if (_sel == this.opcm.selector) _opcm = OPContractsManager(_value);
+        else revert("UpgradeOPCMInput: unknown selector");
+    }
+
+    function set(bytes4 _sel, OPContractsManager.OpChainConfig[] memory _value) public {
```

Comment:
I didn't think these functions could accept anything other than basic types? Ie. why this [getter](https://github.com/ethereum-optimism/optimism/blob/opcm/no-mips32-in-prod/packages/contracts-bedrock/scripts/deploy/DeployOPChain.s.sol#L161-L174) that returns a bytes string. 

Did we change that, or is it OK on setters but not getters?

Reply:
it's a bit nuanced. setters that use complex types will break if they are called from within the script _and_ the script is running in the simulation context. this method is only used in tests, which runs in forge and therefore supports structs.